datasets/jp/meti_eul/crawler.py

```diff
@@ -36,35 +38,70 @@ def crawl_pdf_url(context: Context) -> str:
+                    if len(parts) == 1:
+                        header = parts[0]
+                    elif len(parts) == 2:
+                        header = parts[1]
+                    else:
+                        context.log.error("Unexpected header", header=cell)
+                        return
+                    headers.append(slugify(header, sep="_"))
+                continue
+            assert len(headers) == len(row), (headers, row)
+            yield dict(zip(headers, row))
+
+
 def crawl(context: Context):
     pdf_url = crawl_pdf_url(context)
     cached, path, media_type, charset = fetch_resource(context, "source.pdf", pdf_url)
     if not cached:
         assert media_type == PDF, media_type
+
     context.export_resource(path, PDF, title=context.SOURCE_TITLE)
     last_no = 0
-    for page_path in h.make_pdf_page_images(path):
-        data = run_image_prompt(context, prompt, page_path)
-        assert "rows" in data, data
-        for holder in data.get("rows", []):
-            no = int(holder.get("no"))
-            if no != last_no + 1:
-                context.log.warn(
-                    "Row number is not continuous",
-                    no=no,
-                    last_no=last_no,
-                )
-            last_no = no
-            name = holder.get("name")
-            entity = context.make("LegalEntity")
-            entity.id = context.make_id(str(no), name)
-            entity.add("name", name, lang="eng")
-            entity.add("name", holder.get("name_jpn"), lang="jpn")
-            entity.add("alias", holder.get("also_known_as"))
-            entity.add("country", holder.get("country"))
-            entity.add("topics", "sanction")
-
-            sanction = h.make_sanction(context, entity)
-            sanction.add("reason", holder.get("type_of_wmd"))
-            context.emit(entity, target=True)
-            context.emit(sanction)
+    for holder in parse_pdf_table(context, path):
+        no = int(holder.pop("no"))
+        if no != last_no + 1:
+            context.log.warn(
+                "Row number is not continuous",
+                no=no,
+                last_no=last_no,
+            )
+        last_no = no
+        name = collapse_spaces(holder.pop("company_or_organization")).strip()
+        non_ascii_match = REGEX_NON_ASCII_PARENS.search(name)
+        if non_ascii_match:
+            name_jpn = non_ascii_match.group(1)
+            name = name.replace(name_jpn, "").strip()
+            name_jpn = name_jpn.strip("()（）")
+        else:
+            name_jpn = None
+        for orig, repl in NAME_REPLACEMENTS:
+            name = name.replace(orig, repl)
+
+        aliases = [collapse_spaces(a) for a in holder.pop("also_known_as").split("・")]
+
+        country = collapse_spaces(holder.pop("country_or_region"))
+        country = REGEX_NON_ASCII.sub("", country).strip()
+
+        entity = context.make("LegalEntity")
+        entity.id = context.make_id(str(no), name)
```

Comment:
what do you think of changing no to country? it'll mean lots of rekeying now, but make it more stable in the next update.